plugins/woocommerce-admin/client/homescreen/mobile-app-modal/components/MobileAppLoginStepper.tsx

```diff
@@ -0,0 +1,151 @@
+import { getAdminSetting } from '~/utils/admin-settings';
+import { MobileAppInstallationInfo } from '../components/MobileAppInstallationInfo';
+import { MobileAppLoginInfo } from '../components/MobileAppLoginInfo';
+
+export const MobileAppLoginStepper = ( {
+	step,
+	isJetpackPluginInstalled,
+	wordpressAccountEmailAddress,
+	completeInstallationStepHandler,
+	sendMagicLinkHandler,
+	sendMagicLinkStatus,
+}: {
+	step: 'first' | 'second';
+	isJetpackPluginInstalled: boolean;
+	wordpressAccountEmailAddress: string | undefined;
+	completeInstallationStepHandler: () => void;
+	sendMagicLinkHandler: () => void;
+	sendMagicLinkStatus: SendMagicLinkStates;
+} ) => {
+	const [ stepsToDisplay, setStepsToDisplay ] = useState<
+		StepperProps[ 'steps' ] | undefined
+	>( undefined );
+	// we need to generate one set of steps for the first step, and another set for the second step
+	// because the texts are different after progressing from the first step to the second step
+	useEffect( () => {
+		if ( step === 'first' ) {
+			setStepsToDisplay( [
+				{
+					key: 'first',
+					label: __( 'Install the mobile app', 'woocommerce' ),
+					description: __(
+						'Scan the code below to download or upgrade the app, or visit woocommerce.com/mobile from your mobile device.',
+						'woocommerce'
+					),
+					content: (
+						<>
+							<MobileAppInstallationInfo />
+							<Button
+								className="install-app-button"
+								onClick={ () => {
+									completeInstallationStepHandler();
+								} }
+							>
+								{ __( 'App is installed', 'woocommerce' ) }
+							</Button>
+						</>
+					),
+				},
+				{
+					key: 'second',
+					label: __( 'Sign into the app', 'woocommerce' ),
+					description: '',
+					content: <></>,
+				},
+			] );
+		} else if ( step === 'second' ) {
+			if (
+				isJetpackPluginInstalled &&
+				wordpressAccountEmailAddress !== undefined
+			) {
+				setStepsToDisplay( [
+					{
+						key: 'first',
+						label: __( 'App installed', 'woocommerce' ),
+						description: '',
+						content: <></>,
+					},
+					{
+						key: 'second',
+						label: 'Sign into the app',
+						description: sprintf(
+							/* translators: Reflecting to the user that the magic link has been sent to their WordPress account email address */
+							__(
+								'We’ll send a magic link to %s. Open it on your smartphone or tablet to sign into your store instantly.',
+								'woocommerce'
+							),
+							wordpressAccountEmailAddress
+						),
+						content: (
+							<SendMagicLinkButton
+								isFetching={
+									sendMagicLinkStatus ===
+									SendMagicLinkStates.FETCHING
+								}
+								onClickHandler={ sendMagicLinkHandler }
+							/>
+						),
+					},
+				] );
+			} else {
+				const siteUrl: string = getAdminSetting( 'siteUrl' );
+				const username = getAdminSetting( 'currentUserData' ).username;
+				const loginUrl = `woocommerce://app-login?siteUrl=${ encodeURIComponent(
```

Comment:
This was implemented a bit differently than I had imagined, I was thinking that we could add some code to wc core that can generate the application password, which would be triggered by the user opening this task.

When this modal opens, the QR should already include the application password so that there's no further need for any credentials to be provided by the user.

Furthermore, I was thinking that there should be a deeplink that immediately brings the user to install woocommerce mobile app if they don't already have it, and then when it is installed the app should also have the application password forwarded to it. This would lead to a true seamless experience

Are these part of the future roadmap?

Reply:
Hi @rjchow 👋 

> This was implemented a bit differently than I had imagined, I was thinking that we could add some code to wc core that can generate the application password, which would be triggered by the user opening this task.

This is iteration 2 of the improvements. We have on our radar iteration 3 https://github.com/woocommerce/woocommerce/pull/40628/files, which will include the application password and is closer to what we already discussed.

> Furthermore, I was thinking that there should be a deeplink that immediately brings the user to install woocommerce mobile app if they don't already have it, and then when it is installed the app should also have the application password forwarded to it. This would lead to a true seamless experience

💯 agree with you! We are still looking at how to support deferred deep links in the app, but I agree this is the way to go

Reply:
Ok great to know that! Thanks!